app/models/supplier.rb

```diff
@@ -5,8 +5,22 @@ class Supplier < ApplicationRecord
   validates :name, presence: true
   validates :address, presence: true
   validates :phone_number, presence: true, length: { in: 8..12, allow_blank: true }
+  validates :opens_at, numericality: { in: 0..23 }
+  validates :close_at, numericality: { in: 0..23 }
 
   def delivery?
     delivery
   end
+
+  def self.categories(category)
+    joins(services: [:categories]).where({ categories: { id: category.id } }).group(:name,:id)
+  end
+
+  def self.search(search)
+    if search
+      where('suppliers.name LIKE ?', "%#{search}%")
+    else
+      all
+    end
```

Comment:
another option for the same
`search.present? ? where('suppliers.name LIKE ?', "%#{search}%") : all`